man/man1/p11sak.1.in

```diff
@@ -1,678 +1,1759 @@
+.B CKA_ID
+attribute of the key. Furthermore, the optional
+.I PRIV\-BITS
+argument allows the user to specify the size of the private key in bits.
+.
+.PP
+.SS "Generating DSA keys"
+.
+.B p11sak
+.BR generate\-key | gen\-key | gen
+.BR dsa
+.I DSA\-PARAM\-PEM\-FILE
+.BR \-\-slot | \-s
+.I SLOTID
+.RB [ \-\-pin | \-p
+.IR PIN ]
+.RB [ \-\-force\-pin\-prompt ]
+.BR \-\-label | \-L
+.IR LABEL | PUB\-LABEL : PRIV\-LABEL
+.BR \-\-attr | \-a
+.IR ATTRS | PUB\-ATTRS : PRIV\-ATTRS
+.RB [ \-\-id | \-i
+.IR ID ]
+.RB [ \-\-help | \-h ]
+.PP
+Use the
+.BR generate\-key | gen\-key | gen
+.B dsa
+.I DSA\-PARAM\-PEM\-FILE
+command and
+.I KEYTYPE
+argument to generate a DSA key (\fBCKK_DSA\fP), where
+.I DSA\-PARAM\-PEM\-FILE
+specifies the name of a DSA parameters PEM file. You can for example generate
+DSA parameters using the
+.B OpenSSL
+command line tool as follows: '\fBopenssl dsaparam <numbits> \-out <pem\-file>
+\-outform PEM'\fP. See the OpenSSL man page for details about this command.
+.PP
+The
+.BR \-\-label | \-L
+.IR LABEL | PUB\-LABEL : PRIV\-LABEL
+option sets the
+.B CKA_LABEL
+attribute of the key and the
+.BR \-\-attr | \-a
+.IR ATTRS | PUB\-ATTRS : PRIV\-ATTRS
+option can be used to set the boolean attributes of the key (see below for
+detailed description of the attributes).
+Optionally, set individual key attributes for public and private key separated
+by a colon (\fB:\fP). The
+.BR \-\-id | \-i
+.I ID
+option can be used to set the value of the
+.B CKA_ID
+attribute of the key.
+.PP
 .SS "Generating EC keys"
 .
 .B p11sak
-.BR generate-key | gen-key | gen
+.BR generate\-key | gen\-key | gen
 .BR ec
-.BR CURVE
-.B \-\-slot
-.IR SLOTID
-.B \-\-pin
-.IR PIN
-.B \-\-label
-.IR LABEL
-.B \-\-attr
-.IR [P M R L S E D G V W U A X N T]
-.B \-\-help | \-h
+.I CURVE
+.BR \-\-slot | \-s
+.I SLOTID
+.RB [ \-\-pin | \-p
+.IR PIN ]
+.RB [ \-\-force\-pin\-prompt ]
+.BR \-\-label | \-L
+.IR LABEL | PUB\-LABEL : PRIV\-LABEL
+.BR \-\-attr | \-a
+.IR ATTRS | PUB\-ATTRS : PRIV\-ATTRS
+.RB [ \-\-id | \-i
+.IR ID ]
+.RB [ \-\-help | \-h ]
 .PP
 Use the
-.B generate-key
+.BR generate\-key | gen\-key | gen
 .B ec
-.B CURVE
-command and key argument to generate an EC key, where
 .I CURVE
-specifies the eliptic curve used to create the EC key. The following arguments can be used for respective curves:
-.B prime256v1 | prime192 | secp224 | secp384r1 | secp521r1 | secp265k1 | brainpoolP160r1 | brainpoolP160t1
-.B | brainpoolP192r1 | brainpoolP192t1 | brainpoolP224r1 | brainpoolP224t1 | brainpoolP256r1 | brainpoolP256t1
-.B | brainpoolP320r1 | brainpoolP320t1 | brainpoolP384r1 | brainpoolP384t1 | brainpoolP512r1 | brainpoolP512t1
-.B | curve25519 | curve448 | ed25519 | ed448
+command and
+.I KEYTYPE
+argument to generate an EC key (\fBCKK_EC\fP), where
+.I CURVE
+specifies the elliptic curve used to create the EC key. The following arguments
+can be used for respective curves:
+.BR prime256v1 | prime192 | secp224 | secp384r1 | secp521r1 | secp265k1 | \
```

Comment:
prime192 was prime192v1 in the previous version of p11sak

Reply:
... and secp224 was secp224r1

Reply:
Hm, well there was a discrepancy in the older p11sak version:

Code:
```
    printf("      prime256v1\n");
    printf("      prime192v1\n");
    printf("      secp224r1\n");
    printf("      secp384r1\n");
    printf("      secp521r1\n");
    printf("      secp265k1\n");
```

Man page:

`.SS "prime256v1 | prime192 | secp224 | secp384r1 | secp521r1 | secp265k1 | ....`

So the code always had the v1 or r1 included, but the man page did not. 
What counts is the code....

The new version also has it right in the code:

```
DECLARE_CURVE_INFO(prime256v1, 256);
DECLARE_CURVE_INFO(prime192v1, 192);
DECLARE_CURVE_INFO(secp224r1, 224);
DECLARE_CURVE_INFO(secp384r1, 384);
DECLARE_CURVE_INFO(secp521r1, 521);
DECLARE_CURVE_INFO(secp256k1, 256);
```

... but also wrong in the man page, the same way it was wrong in the old version:

`.BR prime256v1 | prime192 | secp224 | secp384r1 | secp521r1 | secp265k1 | \`

So I'll fix the man page to always use v1/r1, just like it is in the code.